File: 33 - Flatten Deeply Nested Arrays/solution.py
def flatter_array(arr):
    result = []

    for item in arr:
        if isinstance (item, list):
            result = result + flatter_array(item)
        else:
            result.append(item)

    return result

# Test Cases
print(flatter_array([1, [2, [3, [4, 5], 6], 7], 8]))

File: 33 - Flatten Deeply Nested Arrays/solution.js
function flatterArray(arr) {
  let result = [];

  for (let item of arr) {
    if (Array.isArray(item)) {
      result = result.concat(flatterArray(item));
    } else {
      result.push(item);
    }
  }

  return result;
}

// Test Cases
console.log(flatterArray([1, [2, [3, [4, 5], 6], 7], 8]));
